pip/pip-296.md

```diff
@@ -0,0 +1,100 @@
+# Background knowledge
+A consumer is a process that attaches to a topic via a subscription and then receives messages. The reader interface for Pulsar enables applications to manually manage cursors. More knowledge of the Reader and Consumer interface can be found in the [Pulsar Client doc](https://pulsar.apache.org/docs/next/concepts-clients/#reader).
+
+# Motivation
+
+Add the `getLastMessageIds` API for Reader.
+
+It can be used in the following case:
+**Message Tracing and Recovery:**
+In specific scenarios, applications might want to continue consuming from where they left off instead of starting from the beginning. Using the `getLastMessageIds` method, applications can retrieve the list of IDs for the last messages consumed in the previous session. These IDs can then be used during a restart to locate the starting position for consumption.
```

Comment:
Could you explain how do you use `getLastMessageIds` to get the last messages consumed in the previous session?